crates/core/src/operations/subscribe.rs

```diff
@@ -71,9 +71,39 @@ pub(crate) async fn request_subscribe(
         let target = match op_manager.ring.closest_potentially_caching(key, EMPTY) {
             Some(peer) => peer,
             None => {
-                // No remote peers available
-                tracing::debug!(%key, "No remote peers available for subscription");
-                return Err(RingError::NoCachingPeers(*key).into());
+                // No remote peers available - check if we have the contract locally
+                tracing::debug!(%key, "No remote peers available for subscription, checking locally");
+
+                if super::has_contract(op_manager, *key).await? {
+                    // We have the contract locally, subscribe to ourselves
+                    let own_location = op_manager.ring.connection_manager.own_location();
+                    if op_manager
```

Comment:
I am not sure why we need to do this even... we may as well complete the operation. As with all other ops, having self as target for network ops leads to problems more often than not.

I think that this should already be handled by the ws API and all the client request routing and the Session Actor machinery